### DISCUSSION TITLE
Word count

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [buttondown/discussions#16] [Ideas]
Now that I have found Discussions, I can add my wish list items. First up is a live word count. I can see the number of words in the Send summary, but it would be nice to have a visible counter while I write.

**Comment by jmduke** [comment]
I go back and forth on this one for a while; I'm tempted to have it be a hidden toggle. Any other in-writing-experience things you'd want to see at a glance?

**Reply by jdhitsolutions** [reply]
It would be nice to know about approaching the size limit and not discover it when trying to send. I would also appreciate syntax highlighting in code blocks. I can set the language in the markdown code fence, but it doesn't seem to have any effect. I've also mentioned separately the ability to insert a paywall break.

I am also curious about the short line length of what looks like 65. If we have to have a wrap limit, can it be 80? Or is that a style setting I've missed?

**Reply by jmduke** [reply]
Excellent, makes sense. Will mull this a little bit further.

Re: line length — are you referring to in the UI, or within the sent email proper?

**Comment by jdhitsolutions** [comment]
It looks like the line length in the email is about 80. Here's what it looks like in my email client:
![image](https://github.com/buttondown-email/discussions/assets/12089920/909a6faf-03c5-477b-988c-e2db66d802ba)


But in the web archive, the lines are shorter.

![image](https://github.com/buttondown-email/discussions/assets/12089920/c201c23e-fac9-4a93-bdee-92106cf6f782)

To make it more confusing, the email and web previews in the editor show yet another line length.  This is an issue because sometimes I have code samples, and if I know they will wrap, I try to adjust the code layout.

**Reply by jmduke** [reply]
Ahh, I see. We don't automatically _wrap_ in the truncation sense; it's all CSS. For code samples in particular, our default styling does _not_ wrap (we preserve exact whitespace)!

**Reply by jdhitsolutions** [reply]
Ok. I won't obsess over this. I'll adapt the more I write on the platform.
